### PR TITLE
Testcancancan

### PR DESCRIPTION
**Pull request by gandradep**
- Limit access to web app resources based on authorization rules.
- Use cancancan to set authorization permits.
- Admin role user can delete post and comments
- Author can delete post and comments

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -27,4 +27,12 @@ def show
...
     @post = Post.find(params[:id])
   end
+
+  def destroy
+    post = Post.find(params[:id])
+    user = User.find(params[:user_id])
+    post.destroy
+    user.update(postsCounter: user.posts.count)
+    redirect_to user_posts_path(user.id)
+  end
```
Comment (ichala) [inline app/controllers/posts_controller.rb:38]: - [x] to apply CanCanCan features for Posts, kindly add CanCanCan authorization in Post Controller 🙏 

```Ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by ichala — CHANGES_REQUESTED** [review]
Hi Student
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No Linter errors  ✔️ 
- CanCanCan Installed ✔️


 ## Required Changes ♻️


- [ ] **_🛑  Check The Comments Below_**





## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (gandradep) [reply app/controllers/posts_controller.rb:38]: @ichala thanks for the suggestions. I'll fix it right now.

**Review by m05e5 — APPROVED** [review]
**Hi @gandradep 

You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights

✔️ Well done implementing the requested changes 

## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Comment by gandradep** [comment]
@m05e5 thank you for the review.

**Pull request merged**